src/BaseOAuth.php

```diff
@@ -165,7 +171,12 @@ protected function createHttpClient($reference)
      */
     protected function defaultReturnUrl()
     {
-        return Yii::$app->getRequest()->getAbsoluteUrl();
+        $params = Yii::$app->getRequest()->getQueryParams();
+        $params = array_intersect_key($params, array_flip($this->parametersToKeepInReturnUrl));
+
+        $params[0] = Yii::$app->controller->getRoute();
+
+        return Yii::$app->getUrlManager()->createAbsoluteUrl($params);
```

Comment:
I assume the coupling between this extension and specific parts of the app global state is not new and is expected for this kind of ext.

Reply:
Do you mean these components should be passed as parameters? I just copied this part from existing methods.

Reply:
No. Params wouldn't be much better. I was just asking if this kind of coupling was already present before your changes (it was) and is pretty much unavoidable (I think it is).

(It's just a habit I have. in a code review I question when an object makes assumptions about what's in its global environment. For example, I think using `Yii::$app->user` in a web controller is usually ok but I don't like to see that in a model. What if, for example, we want to use the same model in a console command?)